src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -40,71 +42,88 @@ public StatsDPublisher(StatsDConfiguration configuration)
         {
-            _transport.Send(_formatter.Increment(value, bucket));
+            Send(_formatter.Increment(value, bucket));
         }
 
         public void Increment(long value, double sampleRate, string bucket)
         {
-            _transport.Send(_formatter.Increment(value, sampleRate, bucket));
+            Send(_formatter.Increment(value, sampleRate, bucket));
         }
 
         public void Increment(long value, double sampleRate, params string[] buckets)
         {
-            _transport.Send(_formatter.Increment(value, sampleRate, buckets));
+            Send(_formatter.Increment(value, sampleRate, buckets));
         }
 
         public void Decrement(string bucket)
         {
-            _transport.Send(_formatter.Decrement(bucket));
+            Send(_formatter.Decrement(bucket));
         }
 
         public void Decrement(long value, string bucket)
         {
-            _transport.Send(_formatter.Decrement(value, bucket));
+            Send(_formatter.Decrement(value, bucket));
         }
 
         public void Decrement(long value, double sampleRate, string bucket)
         {
-            _transport.Send(_formatter.Decrement(value, sampleRate, bucket));
+            Send(_formatter.Decrement(value, sampleRate, bucket));
         }
 
         public void Decrement(long value, double sampleRate, params string[] buckets)
         {
-            _transport.Send(_formatter.Decrement(value, sampleRate, buckets));
+            Send(_formatter.Decrement(value, sampleRate, buckets));
         }
 
         public void Gauge(long value, string bucket)
         {
-            _transport.Send(_formatter.Gauge(value, bucket));
+            Send(_formatter.Gauge(value, bucket));
         }
 
         public void Gauge(long value, string bucket, DateTime timestamp)
         {
-            _transport.Send(_formatter.Gauge(value, bucket, timestamp));
+            Send(_formatter.Gauge(value, bucket, timestamp));
         }
 
         public void Timing(TimeSpan duration, string bucket)
         {
-            _transport.Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), bucket));
+            Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), bucket));
         }
 
         public void Timing(TimeSpan duration, double sampleRate, string bucket)
         {
-            _transport.Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), sampleRate, bucket));
+            Send(_formatter.Timing(Convert.ToInt64(duration.TotalMilliseconds), sampleRate, bucket));
         }
 
         public void MarkEvent(string name)
         {
-            _transport.Send(_formatter.Event(name));
+            Send(_formatter.Event(name));
+        }
+
+        private void Send(string metric)
+        {
+            try
+            {
+                _transport.Send(metric);
+            }
+            catch (Exception ex)
+            {
+                var handled = _onError(ex);
```

Comment:
There's no guarantee that `_onError` is non-`null` here. Maybe use a construct like this instead?

```csharp
bool handled = _onError?.Invoke(ex) ?? true;
```

Reply:
If we use that code, then the default in `StatsDConfiguration` might as well be null. Seems simpler.